src/Compilers/VisualBasic/Portable/Parser/Parser.vb

```diff
@@ -2945,16 +2945,22 @@ checkNullable:
                 Dim identifierNameOpt As IdentifierNameSyntax = Nothing
                 Dim asKeywordOpt As KeywordSyntax = Nothing
 
+                ' if there is a type character or As, then this must be a name
                 If CurrentToken.Kind = SyntaxKind.IdentifierToken AndAlso
```

Comment:
We should talk to LDM about this. 
"Dim F" does not allow F to be a type so it is always a name, Strict Off just permits omitting the type.

In the tuple case it is the name that is completely optional, while a single Identifier can be a type and, I think, must be a type regardless Strict On or Off. 

---
In reply to: [82047853](https://github.com/dotnet/roslyn/pull/14284#discussion_r82047853) [](ancestors = 82047853)